src/protocols/secure_channel/CheckinMessage.h

```diff
@@ -0,0 +1,90 @@
+/*
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements the Matter Checkin protocol.
+ */
+
+#pragma once
+
+#include <crypto/CHIPCryptoPAL.h>
+#if CHIP_CRYPTO_HSM
+#include <crypto/hsm/CHIPCryptoPALHsm.h>
+#endif
+#include <lib/support/Base64.h>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeDelegate.h>
+#include <messaging/ExchangeMessageDispatch.h>
+#include <protocols/secure_channel/Constants.h>
+#include <protocols/secure_channel/PairingSession.h>
+#include <protocols/secure_channel/SessionEstablishmentExchangeDispatch.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/CryptoContext.h>
+#include <transport/raw/MessageHeader.h>
+#include <transport/raw/PeerAddress.h>
+
+namespace chip {
+namespace Protocols {
+namespace SecureChannel {
+using namespace Crypto;
+
+#define CHIP_CHECK_IN_APP_DATA_MAX_SIZE 1024
```

Comment:
Why is this a define, not a constexpr (on the class?) with a nicely readable name?

More importantly, where is this value coming from?  I don't see this constraint in the spec....  Shouldn't we be allowing all payload sizes up to whatever the packet size allows?

Reply:
Or we could simply remove it since the called is providing the Buffer, but however we would face some allocation problem upon reception as to determined what is the actual size of the appData.

Reply:
The right limit should presumably be "1280 bytes minus IP/UDP headers minus Matter headers" or so....

Reply:
Issue filed #28603